cmd/ci-operator-prowgen/main.go

```diff
@@ -181,13 +194,16 @@ func main() {
 	}
 	genJobs := generateJobsToDir(opt.toDir)
 	for _, subDir := range args {
-		if err := config.OperateOnCIOperatorConfigSubdir(opt.fromDir, subDir, genJobs); err != nil {
-			fields := logrus.Fields{"target": opt.toDir, "source": opt.fromDir, "subdir": subDir}
+		if err := opt.OperateOnCIOperatorConfigDir(filepath.Join(opt.fromDir, subDir), genJobs); err != nil {
+			fields := logrus.Fields{"target": opt.toDir, "source": opt.fromDir}
 			logrus.WithError(err).WithFields(fields).Fatal("Failed to generate jobs")
 		}
-		if err := pruneStaleJobs(opt.toDir, subDir); err != nil {
-			fields := logrus.Fields{"target": opt.toDir, "source": opt.fromDir, "subdir": subDir}
-			logrus.WithError(err).WithFields(fields).Fatal("Failed to prune stale generated jobs")
+
+		if opt.ProcessAll() {
```

Comment:
Are we sure this is always going to do the right thing? There's no case where the user won't have pruning run but needs to?

Reply:
The problem is that the logic that we do the pruning doesn't fit with the current changes. For now, I would focus on pruning only when we process all the files. In the future, I can refactor the whole thing.